File: src/main/java/net/kingdommc/darkages/numinoustreasury/recipe/NuminousRecipeService.java
package net.kingdommc.darkages.numinoustreasury.recipe;

import com.rpkit.core.service.Service;
import net.kingdommc.darkages.numinoustreasury.NuminousTreasury;
import org.bukkit.configuration.file.YamlConfiguration;

import java.io.File;
import java.util.Arrays;
import java.util.List;

public final class NuminousRecipeService implements Service {

    private final NuminousTreasury plugin;
    private final List<NuminousRecipe> recipes;

    public NuminousRecipeService(NuminousTreasury plugin) {
        this.plugin = plugin;
        File recipesFolder = new File(plugin.getDataFolder(), "recipes");
        if (!recipesFolder.exists()) {
            recipesFolder.mkdirs();
        }
        this.recipes = Arrays.stream(recipesFolder.listFiles()).map(recipeFile -> {
            YamlConfiguration itemConfiguration = YamlConfiguration.loadConfiguration(recipeFile);
            return itemConfiguration.getObject("recipe", NuminousRecipe.class);
        }).toList();
    }

    @Override
    public NuminousTreasury getPlugin() {
        return plugin;
    }

    public List<NuminousRecipe> getRecipes() {
        return recipes;
    }
}

File: src/main/kotlin/net/arvandor/numinoustreasury/recipe/NuminousRecipeService.kt
package net.arvandor.numinoustreasury.recipe

import com.rpkit.core.service.Service
import net.arvandor.numinoustreasury.NuminousTreasury
import org.bukkit.configuration.file.YamlConfiguration
import java.io.File

class NuminousRecipeService(private val plugin: NuminousTreasury) : Service {
    val recipes: List<NuminousRecipe>

    init {
        val recipesFolder = File(plugin.dataFolder, "recipes")
        if (!recipesFolder.exists()) {
            recipesFolder.mkdirs()
        }
        this.recipes =
            recipesFolder.listFiles().mapNotNull { recipeFile ->
                val itemConfiguration = YamlConfiguration.loadConfiguration(recipeFile)
                val recipe = itemConfiguration.getObject("recipe", NuminousRecipe::class.java)
                if (recipe == null) {
                    plugin.logger.warning("Failed to load recipe from " + recipeFile.name)
                    return@mapNotNull null
                }
                return@mapNotNull recipe
            }
        plugin.logger.info("Loaded " + recipes.size + " recipes")
    }

    override fun getPlugin(): NuminousTreasury {
        return plugin
    }
}
